shell/egl_window.h

```diff
@@ -67,22 +72,30 @@ class EglWindow : public Egl {
 
   size_t m_index;
   std::shared_ptr<Display> m_display;
+  std::shared_ptr<Engine> m_flutter_engine;
 
   struct wl_surface* m_surface;
-  struct wl_shell_surface* m_shell_surface{};
   wl_egl_window* m_egl_window[kEngineInstanceCount]{};
 
-  int32_t m_width;
-  int32_t m_height;
-
   bool m_fullscreen;
-  bool m_sprawl;
+  bool m_maximized{};
+  bool m_resize{};
+  bool m_activated{};
+  bool m_running{};
+  struct {
+    int32_t width;
+    int32_t height;
+  } m_geometry;
```

Comment:
`m_geometry, m_window_size{}`?

Reply:
Optional window decorations are next hence window vs geometry.

Reply:
Yeah, I was just hinting that we can use the same struct  here instead of declaring a new anonymous one.